ranger/ext/human_readable.py

```diff
@@ -8,52 +8,49 @@
 from ranger.core.shared import SettingsAware
 
 
-def human_readable(byte, separator=' '):  # pylint: disable=too-many-return-statements
+def human_readable(byte_count, separator=' '):  # pylint: disable=too-many-return-statements
     """Convert a large number of bytes to an easily readable format.
+    Output depends on the binary_size_prefix setting, which is False by default.
 
     >>> human_readable(54)
     '54 B'
     >>> human_readable(1500)
-    '1.46 K'
+    '1.5 k'
     >>> human_readable(2 ** 20 * 1023)
-    '1023 M'
+    '1.07 G'
+    >>> SettingsAware.settings.binary_size_prefix = True
+    >>> human_readable(54)
+    '54 B'
+    >>> human_readable(1500)
+    '1.46 Ki'
+    >>> human_readable(2 ** 20 * 1023)
+    '1023 Mi'
     """
 
     # handle automatically_count_files false
-    if byte is None:
+    if byte_count is None:
         return ''
-
+    if byte_count <= 0:
+        return '0'
     if SettingsAware.settings.size_in_bytes:
-        return format(byte, 'n')  # 'n' = locale-aware separator.
+        return format(byte_count, 'n')  # 'n' = locale-aware separator.
 
-    # I know this can be written much shorter, but this long version
-    # performs much better than what I had before.  If you attempt to
-    # shorten this code, take performance into consideration.
-    if byte <= 0:
-        return '0'
-    if byte < 2**10:
-        return '%d%sB' % (byte, separator)
-    if byte < 2**10 * 999:
-        return '%.3g%sK' % ((byte / 2**10), separator)
-    if byte < 2**20:
-        return '%.4g%sK' % ((byte / 2**10), separator)
-    if byte < 2**20 * 999:
-        return '%.3g%sM' % ((byte / 2**20), separator)
-    if byte < 2**30:
-        return '%.4g%sM' % ((byte / 2**20), separator)
-    if byte < 2**30 * 999:
-        return '%.3g%sG' % ((byte / 2**30), separator)
-    if byte < 2**40:
-        return '%.4g%sG' % ((byte / 2**30), separator)
-    if byte < 2**40 * 999:
-        return '%.3g%sT' % ((byte / 2**40), separator)
-    if byte < 2**50:
-        return '%.4g%sT' % ((byte / 2**40), separator)
-    if byte < 2**50 * 999:
-        return '%.3g%sP' % ((byte / 2**50), separator)
-    if byte < 2**60:
-        return '%.4g%sP' % ((byte / 2**50), separator)
-    return '>9000'
+    # If you attempt to shorten this code, take performance into consideration.
+    binary = bool(SettingsAware.settings.binary_size_prefix)
```

Comment:
Can we maybe turn this into a default argument to the function? That way it'd be easier to override if need be.

Reply:
I tried it, and it threw an error `type object 'SettingsAware' has no attribute 'settings'`. That's probably due to the fact that when the method is defined and the default arguments are evaluated, `SettingsAware` has not yet had its method `.settings_set` called (so it doesn't contain the `.settings` attribute). I'm not an expert in the codebase, so there might be something different going on. I'm not sure.

Reply:
Ah, darn, then I'd say to go ahead with `None` as the default argument and checking with `is None`.

Reply:
@toonn I pushed the changes, but forgot about the "line too long" pylint error. Maybe I should just turn this long conditional expression into a proper if statement?
```python
binary = use_binary if use_binary is not None else bool(SettingsAware.settings.binary_size_prefix)
```
Or maybe just omit the `bool()` call?

Reply:
I believe we can drop the `bool()`. If you declare the setting as bool Ranger should take care of converting regardless of case and I believe yes/no should also work.

And I think flipping the conditional is also slightly clearer.

`binary = SettingsAware.settings.binary_size_prefix if use_binary is None else use_binary`